File: designpatterns/src/Command/Light.java
package Command;

public class Light {
    String place;

    public Light() { }

    public Light(String place) {
        this.place = place;
    }

    public void on() {
        if(place != null) {
            System.out.println(place + " light is on");
        } else {
            System.out.println("Light is on");
        }
    }

    public void off() {
        if(place != null) {
            System.out.println(place + " light is off");
        } else {
            System.out.println("Light is off");
        }
    }
}

File: designpatterns/kotlin/designpatterns/src/main/kotlin/command/example/Light.kt
package command.example

class Light {

    fun on() {
        println("Light is on")
    }

    fun off() {
        println("Light is off")
    }
}

class LightOnCommand(
    val light: Light,
): Command {
    override fun execute() {
        light.on()
    }

    override fun undo() {
        light.off()
    }
}

class LightOffCommand(
    val light: Light,
): Command {
    override fun execute() {
        light.off()
    }

    override fun undo() {
        light.on()
    }

}